### PR TITLE
Preparing to release UPM Packages to Unity's Asset store. 

### PR DESCRIPTION
**Pull request by AMollis**
This PR updates MRTK packages so that they can be published to the Unity Asset Store.  Doing this required the following changes:

1. Adding `unityRelease` and `documentationUrl` fields to the package.json files. 
2. Adding CHANGELOG.md with correct versioning.  After this change PRs must add notable changes to the package's CHANGELOG.md file.
3. Converted one JPG image to PNG.  Unity Asset Store packages can't contain compressed images.
4. Updated package versions where needed, to reflect changes made after pending release (see #628)
5. Added unit tests to validate packages using Unity's "Asset Store Validation" tool.
6. Updated build scripts to update package's prerelease tags and dates in CHANGELOG.md to the correct value.
7. Updated sample dependencies to the latest versions that also support the Unity Asset Store.  In a future change we should update the sample to use the Unity Asset Store directly. 

‼️ Note, the min Unity version supporting UPM Asset Store packages is 2021.3.26f1.  That's why you see `"unityRelease"="26f1"`.  This change won't prohibit people from obtaining the MRTK packages via the MR Feature Tool for older versions of the Unity.   

## Depends On
- https://github.com/microsoft/MixedReality-GraphicsTools-Unity/pull/195
- https://github.com/microsoft/MixedRealityToolkit/pull/304

## Changes
- Fixes: https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/issues/518

### CODE REVIEW

**Comment by AMollis** [comment]
@MixedRealityToolkit/mixedrealitytoolkit-unity-maintainers  This PR is ready for a review

File: org.mixedrealitytoolkit.tools/Tests/Editor/MRTK.Tools.EditorTests.asmdef [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "autoReferenced": false,
+    "defineConstraints": [],
+    "versionDefines": [
+        {
+            "name": "com.unity.asset-store-validation",
+            "expression": "",
+            "define": "HAS_ASSET_STORE_VALIDATION"
+        },
+        {
+            "name": "org.mixedrealitytoolkit.core",
```
Comment (keveleigh) [inline org.mixedrealitytoolkit.tools/Tests/Editor/MRTK.Tools.EditorTests.asmdef:27]: Why would these packages ever _not_ have the Core package installed alongside, when they have the Core package as a dependency?

https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/6bc4be65047cd3f65f608f2b6e406e1c31d72527/org.mixedrealitytoolkit.tools/package.json#L20

File: org.mixedrealitytoolkit.input/Tests/Editor/PackageValidationTest.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clause
+
+// Disable "missing XML comment" warning for tests. While nice to have, this documentation is not required.
```
Comment (keveleigh) [inline org.mixedrealitytoolkit.input/Tests/Editor/PackageValidationTest.cs:4]: nit: I personally think inserting some standard, consistent, simple XML headers for the tests looks cleaner than disabling a warning for each file.

Comment (AMollis) [reply org.mixedrealitytoolkit.input/Tests/Editor/PackageValidationTest.cs:4]: I can do this for you @keveleigh  :)  For the new stuff at least.

File: org.mixedrealitytoolkit.tools/Tests/Editor/MRTK.Tools.EditorTests.asmdef (hunk 1 above)
Comment (AMollis) [reply org.mixedrealitytoolkit.tools/Tests/Editor/MRTK.Tools.EditorTests.asmdef:27]: Ahh....you caught me!

I copied and pasted across all packages for consistency.  Not all packages directly depend on "core" directly.

Seems like you have an opinion here, so I'll remove the conditions for packages that have Core as a direct dependency, but will leave the others.

**Comment by AMollis** [comment]
@keveleigh  resolved your two comments. Thank you

File: org.mixedrealitytoolkit.windowsspeech/Tests/Editor/PackageValidationTest.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    internal class PackageValidationTest
+    {
+        /// <summary>
+        /// Test to validate the package for the Mixed Reality Toolkit Windows Speech package, verifying that all
+        /// requirements are met for publishing the package to the Unity Asset Store.
+        /// </summary>
+        [Test]
+        public void PackageTest()
+        {
+            PackageValidatorResults results = PackageValidator.Validate("org.mixedrealitytoolkit.windowsspeech");
```
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.windowsspeech/Tests/Editor/PackageValidationTest.cs:24]: Nit:  "org.mixedrealitytoolkit.windowsspeech" could be stored in a constant string as in: `const string PackageName="org.mixedrealitytoolkit.windowsspeech"` as a good practice.
Note: This suggestion applies to all PackageValidationTest.cs in this PR

File: org.mixedrealitytoolkit.windowsspeech/Tests/Editor/PackageValidationTest.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            Assert.AreEqual(0, results.FailedCount, $"Failed tests found.\n{results.ToString(PackageValidatorResults.MessageType.Failed)}");
```
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.windowsspeech/Tests/Editor/PackageValidationTest.cs:25]: Nit:  Replace `\n` with Environment.NewLine as a portability good practice.
Note: This suggestion applies to all PackageValidationTest.cs files in this PR.

File: org.mixedrealitytoolkit.core/Tests/TestUtilities/PackageValidatorResults.cs [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+        /// <summary>
+        /// The number of succeeded tests.
+        /// </summary>
+        public int SucceededCount { get; private set; }
+
+        /// <summary>
+        /// The number of tests that were not run.
+        /// </summary>
+        /// <value></value>
```
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.core/Tests/TestUtilities/PackageValidatorResults.cs:59]: Not needed, typo perhaps?

File: org.mixedrealitytoolkit.core/Tests/TestUtilities/PackageValidatorResults.cs [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+
+        /// <summary>
+        /// Load the results from the package validation log.
+        /// </summary>
+        /// <exception cref="FileNotFoundException">
+        /// Thrown if the package's log file is not found.
+        /// </exception>
+        public void LoadResults()
+        {
+            string filePath = Path.Combine(Application.dataPath, "..", directoryPath, fileName);
```
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.core/Tests/TestUtilities/PackageValidatorResults.cs:102]: Nit:  `".."` could be a class level const string as a coding good practice.

File: org.mixedrealitytoolkit.core/Tests/TestUtilities/PackageValidatorResults.cs [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException($"File not found: {filePath}");
+            }
+
+            string fileContent = File.ReadAllText(filePath);
+            var startIndex = fileContent.IndexOf("VALIDATION RESULTS:");
```
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.core/Tests/TestUtilities/PackageValidatorResults.cs:109]: `"VALIDATION RESULTS:"` is repeated throughout the code, better place it in a class-level `const string ValidationResultsCaption="VALIDATION RESULTS:"` or something similar, to facilitate conde maintenance.

File: org.mixedrealitytoolkit.standardassets/Textures/Images/Person.png
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.standardassets/Textures/Images/Person.png:1]: Just double-checking:  Can we redistribute this person's picture?

File: org.mixedrealitytoolkit.windowsspeech/Tests/Editor/PackageValidationTest.cs (hunk 3 above)
Comment (keveleigh) [reply org.mixedrealitytoolkit.windowsspeech/Tests/Editor/PackageValidationTest.cs:24]: The compiler is pretty good at detecting which strings are const. I put together some quick tests at [sharplab.io](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQGYACLEgYRIG8CT6zSBjAexgGcAXM7ABhIAqAUy4kAvCQBELAE4BzAHRgoADyEIZQiABsonAJ6cWLbQGs9CgO6wELS+3YAHIUKYALSQG46DYjwBsJJoQtjDa+jz8wlyY4lKyispqGlq6BkYm5pxWNnYOzq4e3vgMjJGCIpykEtLySqrqwWmGxmYW1jC29k4u7l4+9ANl6AAsJACyABQAlDRDpejYAJyTtYkNKTp6LZntud0FfdPFpQvLk9Gcx/O+55eY1yWn9Isrl0THz4NPz6+TACSSS6wOQgGiXAC+kkeXx4K0BwJgoPBlUwUJhXz+CMqILB1He6JODAhBAhQA) that show how the compiler treats these cases, and refactoring out as "const" appears equivalent both from the C# view (the consts get inlined anyway) and the IL view. You can switch between views from this dropdown:

image

File: org.mixedrealitytoolkit.standardassets/Textures/Images/Person.png
Comment (AMollis) [reply org.mixedrealitytoolkit.standardassets/Textures/Images/Person.png:1]: This picture is before my time.   I'm just convert from JPG to PNG

@keveleigh  do you know where this is fromt?

File: org.mixedrealitytoolkit.windowsspeech/Tests/Editor/PackageValidationTest.cs (hunk 3 above)
Comment (AMollis) [reply org.mixedrealitytoolkit.windowsspeech/Tests/Editor/PackageValidationTest.cs:24]: Leaving as is given @keveleigh 's statement.   Thanks for testing that out @keveleigh .

File: org.mixedrealitytoolkit.standardassets/Textures/Images/Person.png
Comment (keveleigh) [reply org.mixedrealitytoolkit.standardassets/Textures/Images/Person.png:1]: I _think_ @cre8ivepark originally contributed it, but I don't know much more.

File: org.mixedrealitytoolkit.core/Tests/TestUtilities/PackageValidatorResults.cs [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+            if (startIndex == -1) throw new Exception("File does not contain 'VALIDATION RESULTS:'");
+
+            var resultsContent = fileContent.Substring(startIndex + "VALIDATION RESULTS:".Length);
+
+            var failedPrefix = "\nFailed - ";
+            var succeededPrefix = "\nSucceeded - ";
+            var notRunPrefix = "\nNotRun - ";
```
[the remark this answers is not in the document]
Comment (AMollis) [reply org.mixedrealitytoolkit.core/Tests/TestUtilities/PackageValidatorResults.cs]: Unfortunately I can't use `Environment.NewLine` here.  On Windows that is "\r\n", while the log file only uses Unix style  new lines, "\n".

I've added a code comment explaining this

File: org.mixedrealitytoolkit.standardassets/Textures/Images/Person.png
Comment (AMollis) [reply org.mixedrealitytoolkit.standardassets/Textures/Images/Person.png:1]: I've opened this bug to track this issue:

- https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/issues/634

**Review by shaynie — APPROVED** [review]

**Review by shaynie — APPROVED** [review]

**Pull request merged**